### PR TITLE
Score Observer, Game Observable, Text Class, Name TextField

### PR DESCRIPTION
**Pull request by mauricewillemsen**
Set Score Class Observable, Set Game,Pause-,Defeat-,VictoryScreen Observer. 
Added Text Class (based on Button) class to show text on screens.
Added Textfield to MainMenu to set PlayerName.
Added some Javadoc

### CODE REVIEW

**Comment by peace-4-you** [comment]
Are we fixing mavin plugin errors immediately on new codes? Mavin plugin = checkstyle, pmd,.....

**Comment by mauricewillemsen** [comment]
Fixed most checkstyle errors now

**Comment by peace-4-you** [comment]
Why is there a new score variable in Game class? What is the idea behind it ?

File: bust-a-move-framework/src/main/java/nl/tudelft/bust_a_move20162017/bust_a_move_framework/gamestate/GameConfig.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	 */
+	public static final int WIDTH1 = 100;
+	/**
+	 * Width of something to display on the screen.
+	 */
+	public static final int WIDTH2 = 150;
+	/**
+	 * Width of something to display on the screen.
+	 */
+	public static final int WIDTH3 = 200;
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/nl/tudelft/bust_a_move20162017/bust_a_move_framework/gamestate/GameConfig.java]: Maybe WIDTH200 is a more descriptive name

File: bust-a-move-framework/src/main/java/nl/tudelft/bust_a_move20162017/bust_a_move_framework/gamestate/GameConfig.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	 */
+	public static final int SEVENTH_LINE = 280;
+	/**
+	 * Width of something to display on the screen.
+	 */
+	public static final int WIDTH1 = 100;
+	/**
+	 * Width of something to display on the screen.
+	 */
+	public static final int WIDTH2 = 150;
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/nl/tudelft/bust_a_move20162017/bust_a_move_framework/gamestate/GameConfig.java]: Maybe WIDTH150 is a more descriptive name

File: bust-a-move-framework/src/main/java/nl/tudelft/bust_a_move20162017/bust_a_move_framework/gamestate/GameConfig.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	 */
+	public static final int SIXT_LINE = 240;
+	/**
+	 * Vertical Position of the seventh line to display on the screen.
+	 */
+	public static final int SEVENTH_LINE = 280;
+	/**
+	 * Width of something to display on the screen.
+	 */
+	public static final int WIDTH1 = 100;
```
Comment (peace-4-you) [inline bust-a-move-framework/src/main/java/nl/tudelft/bust_a_move20162017/bust_a_move_framework/gamestate/GameConfig.java]: Maybe WIDTH100 is a more descriptive name. What do you think ?

**Comment by defnoob** [comment]
With regards to the width of the buttons, I can change the button class to be more adaptable to the width of the text. If you want all buttons in the same gamestate to be the same width however...(could probably still do it with a workaround).

**Comment by peace-4-you** [comment]
> With regards to the width of the buttons, I can change the button class to be more adaptable to the width of the text. If you want all buttons in the same gamestate to be the same width however...(could probably still do it with a workaround).

I don't think it's a large enough problem to justify your suggested change.


> Edit: I checked the rest of the game config file and it would probably be better if there a group of buttons with an initial starting y, since each button has a fixed y-offset between them.

Isn't that what the FIRST_LINE, SECOND_LINE, .... is for ?

**Comment by defnoob** [comment]
Well in that case I've nothing else to point out. This gets the go from me.

**Pull request merged**